apps/coscrad-frontend/src/utils/generic-components/presenters/tables/generic-index-table-presenter.tsx

```diff
@@ -0,0 +1,76 @@
+import { IBaseViewModel } from '@coscrad/api-interfaces';
+import './generic-index-table-presenter.css';
+
+// TODO add me to the types library
+export type ValueUnion<T> = T[keyof T];
+
+export type Renderer<T> = (row: T) => JSX.Element | string;
+
+export type CellRenderers<T extends IBaseViewModel> = Map<keyof T, Renderer<T>>;
+
+export type HeadingLabel<T extends IBaseViewModel> = {
+    propertyKey: keyof T;
+    headingLabel: string;
+};
+
+const buildDefaultRenderer =
```

Comment:
We need to break the utility into its own directory, presumably with separate files for several of these pieces.

Reply:
done